### PR TITLE
feat(baremetal): add wait method for server private networks

### PR DESCRIPTION
**Pull request by yfodil**

### CODE REVIEW

File: api/baremetal/v1/server_utils.go [hunk 1]
```diff
@@ -207,3 +207,58 @@ func (s *API) WaitForServerOptions(req *WaitForServerOptionsRequest, opts ...scw
...
+	}
+	retryInterval := defaultRetryInterval
+	if req.RetryInterval != nil {
+		retryInterval = *req.RetryInterval
+	}
+
+	terminalStatus := map[ServerPrivateNetworkStatus]struct{}{
+		ServerPrivateNetworkStatusAttached: {},
+		ServerPrivateNetworkStatusError:    {},
+		ServerPrivateNetworkStatusUnknown:  {},
```
Comment (Monitob) [inline api/baremetal/v1/server_utils.go:234]: Is this a terminal state? can you confirm that with the team?

File: api/baremetal/v1/server_utils.go [hunk 2]
```diff
@@ -207,3 +207,58 @@ func (s *API) WaitForServerOptions(req *WaitForServerOptionsRequest, opts ...scw
...
+type WaitForServerPrivateNetworksRequest struct {
+	ServerID      string
+	Zone          scw.Zone
+	Timeout       *time.Duration
+	RetryInterval *time.Duration
+}
+
+// WaitForServerPrivateNetworks wait for all server private networks to be in a "terminal state" before returning.
+// This function can be used to wait for all server private networks to be set.
+func (s *PrivateNetworkAPI) WaitForServerPrivateNetworks(req *WaitForServerPrivateNetworksRequest, opts ...scw.RequestOption) (*ServerPrivateNetwork, error) {
```
Comment (Monitob) [inline api/baremetal/v1/server_utils.go:221]: Please modify the response to a list of private networks.

File: api/baremetal/v1/server_utils.go [hunk 3]
```diff
@@ -207,3 +207,58 @@ func (s *API) WaitForServerOptions(req *WaitForServerOptionsRequest, opts ...scw
...
+			return res, true, err
+		},
+		Timeout:          timeout,
+		IntervalStrategy: async.LinearIntervalStrategy(retryInterval),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "waiting for server private networks failed")
+	}
+
+	return serverPrivateNetwork.(*ServerPrivateNetwork), nil
```
Comment (Monitob) [inline api/baremetal/v1/server_utils.go:263]: You should return `ListServerPrivateNetworksResponse`

**Review by Monitob — CHANGES_REQUESTED** [review]

File: api/baremetal/v1/server_utils.go (hunk 1 above)
Comment (yfodil) [reply api/baremetal/v1/server_utils.go:234]: Yes it is

**Review by remyleone — APPROVED** [review]

**Review by Monitob — APPROVED** [review]

**Pull request merged**